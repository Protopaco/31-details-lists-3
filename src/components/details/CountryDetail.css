.figure{
    width: 800px;
    height: 700px;
    border: 2px solid brown;
    border-radius: 15px;
    background-color: beige;
}

.img{
    width: 750px;
    height: 500px;
    padding: 25px;
    border-bottom: 4px solid brown;
}

.caption{
    display: flex;
    flex-direction: column;
    /* justify-content: center; */
    align-items: center;
    margin: 0px; 
    /* padding: 0px; */
}

p{
    margin: 10px;
    font-size: 1.5em; 
    color: brown;
    font-weight: bold;
}

.frame{
    width: 100vw;
    height: 100vh - 200px;
    display:flex;
    justify-content: center;
}