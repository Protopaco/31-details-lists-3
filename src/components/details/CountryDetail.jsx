import React from 'react';
import PropTypes from 'prop-types';
import styles from './CountryDetail.css'

const CountryDetail = ({ name, nativeName, flagURL, capital }) => (
    <div className={styles.frame} >
        <figure className={styles.figure}>
            <img className={styles.img} src={flagURL} alt={name} />
            <figcaption className={styles.caption}>
                <p>{name}</p>
                <p>{nativeName}</p>
                <p>{capital}</p>
            </figcaption>
        </figure>
    </div>
);

CountryDetail.propTypes = {
    name: PropTypes.string.isRequired,
    nativeName: PropTypes.string.isRequired,
    flagURL: PropTypes.string.isRequired,
    capital: PropTypes.string.isRequired
};

export default CountryDetail;
