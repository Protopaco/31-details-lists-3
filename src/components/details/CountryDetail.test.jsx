import React from 'react';
import { BrowserRouter as Router } from 'react-router-dom'
import { render, cleanup } from '@testing-library/react';
import CountryDetail from './CountryDetail';

describe('CountryDetail component', () => {
    afterEach(() => cleanup());
    it('renders CountryDetail', () => {
        const { asFragment } = render(
            <Router>
                <CountryDetail
                    name={'f'}
                    nativeName={'f'}
                    flagURL={'f'}
                    capital={'f'}
                />
            </Router>);
        expect(asFragment()).toMatchSnapshot();
    });
});
