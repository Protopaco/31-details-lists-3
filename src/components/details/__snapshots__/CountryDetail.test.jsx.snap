// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CountryDetail component renders CountryDetail 1`] = `
<DocumentFragment>
  <div
    class="frame"
  >
    <figure
      class="figure"
    >
      <img
        alt="f"
        class="img"
        src="f"
      />
      <figcaption
        class="caption"
      >
        <p>
          f
        </p>
        <p>
          f
        </p>
        <p>
          f
        </p>
      </figcaption>
    </figure>
  </div>
</DocumentFragment>
`;
